### PR TITLE
Update pegs.nim to work at compiletime. No range errors.

### PR DESCRIPTION
**Pull request by solo989**

### CODE REVIEW

**Comment by krux02** [comment]
test case is missing. The best way to test this is to transform an already existing pegs test into a test that runs both at run time as well as at compile time.

**Comment by solo989** [comment]
Added a test case.

File: lib/pure/pegs.nim [hunk 1]
```diff
@@ -2040,6 +2058,8 @@ proc escapePeg*(s: string): string =
 when isMainModule:
   assert escapePeg("abc''def'") == r"'abc'\x27\x27'def'\x27"
   assert match("(a b c)", peg"'(' @ ')'")
+  static:
+    assert match("(a b c)", peg"'(' @ ')'")
```
Comment (krux02) [inline lib/pure/pegs.nim:2061]: Here is an example how to lift a test to be executed at compile time as well as runtime:
https://github.com/nim-lang/Nim/blob/devel/tests/stdlib/tjsonmacro.nim#L638

**Review by krux02 — CHANGES_REQUESTED** [review]

**Comment by solo989** [comment]
Lifted all tests.

**Review by krux02 — APPROVED** [review]

**Pull request merged**